File: Projetos_AfonsoMarcondes/C++/0809/Exer18.c
#include <stdio.h>

int func(int num){
    if(num < 1){
        return 1;
    }
    return num * func(num - 1);

}

int main(){
    int num1;
    int resultado;
    printf("Insira um numero: ");
    scanf("%d", &num1);

    resultado = func(num1);
    printf("O resultado é %d", resultado);

}

File: Projetos_AfonsoMarcondes/UFCD 10789/Python/Exercicios/Exer18.py
num = int(input("Insira um numero: "))
lista = []
soma = 0

for i in range(1, num):
    if num % i == 0:
        lista.append(i)
    
soma = sum(lista)
print("Os seus divisores são",lista, "\nA soma deles dá", soma)

if soma == num:
    print("Então é um numero **PERFEITO**")
else:
    print("Então não é um numero perfeito.")
